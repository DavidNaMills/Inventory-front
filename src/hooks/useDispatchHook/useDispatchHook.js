import { useDispatch } from 'react-redux';

import { userLogin, userLogout, changeCurrentLocation } from '../../store/userStore/userActions';
import { startInitTypes, addNewProductType, addNewLocation, removeLocationType } from '../../store/typesStore/typesStoreActions';
import {createMessage} from '../../store/messagingStore/messagingActions';

const useDispatchHook = () => {
    const dispatch = useDispatch();

    const userLoginDispatch = (userData) => dispatch(userLogin(userData));
    const userLogoutDispatch = () => dispatch(userLogout());
    const changeCurrentLocationDispatch = (id) => dispatch(changeCurrentLocation(id));

    const initTypesDispatch = () => dispatch(startInitTypes());
    const addNewProductDispatch = (newType) => dispatch(addNewProductType(newType));
    const addNewLocationDispatch = (newLocation) => dispatch(addNewLocation(newLocation));
    const removeLocationDispatch = (id) => dispatch(removeLocationType(id));

    const createMessageDispatch = (data) => dispatch(createMessage(data));

    return {
        userLoginDispatch,
        userLogoutDispatch,
        changeCurrentLocationDispatch,

        initTypesDispatch,
        addNewProductDispatch,
        addNewLocationDispatch,
        removeLocationDispatch,

        createMessageDispatch
    }
}

export default useDispatchHook;