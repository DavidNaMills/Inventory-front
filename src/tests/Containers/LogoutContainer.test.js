import React from 'react';
import '../testHelpers/context';
import './containerHelper';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import Logout from '../../Containers/LogoutContainer/LogoutContainer';
import { Redirect, BrowserRouter } from 'react-router-dom';

jest.mock('../../hooks/useDispatchHook/useDispatchHook');
import useDispatchHook from '../../hooks/useDispatchHook/useDispatchHook';

const mockLogoutDispatch = jest.fn();

describe('<LogoutContainer/> component test suite', () => {
    let wrapper;

    beforeEach(() => {
        useDispatchHook.mockImplementation(() => ({
            userLogoutDispatch: mockLogoutDispatch
        }));
        wrapper = mount(
            <BrowserRouter>
                <Logout />
            </BrowserRouter>
        );
    });

    it('matches snapshot', () => {
        const comp = renderer.create(
            <BrowserRouter>
                <Logout />
            </BrowserRouter>
        );
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });

    it('exists', () => {
        expect(wrapper.exists).toBeTruthy();
    });

    it('renders a <Redirect/> component', () => { 
        expect(wrapper.find(Redirect).length).toBeTruthy();
        expect(wrapper.find(Redirect).props()).toHaveProperty('to', '/');
    });

    it('invokes userLogoutDispatch() on render', () => { 
        expect(mockLogoutDispatch).toHaveBeenCalledTimes(1);
    });
});