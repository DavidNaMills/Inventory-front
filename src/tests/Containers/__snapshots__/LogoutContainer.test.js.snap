// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoutContainer/> component test suite matches snapshot 1`] = `null`;
